LIMS2DB/classes.py

```diff
@@ -1042,3 +1043,51 @@ def extract_barcode(self, chain):
             if matches:
                 barcode = matches.group(0).replace('_','-')
         return barcode
+
+    def set_status(self):
+        proj_details = self.obj.get('details')
+        status_fields = {}
+
+        # Convenience string status field
+        status_fields['status'] = None
+
+        # Boolean status fields
+        status_fields['aborted'] = False
+        status_fields['closed'] = False
+        status_fields['ongoing'] = False
+        status_fields['open'] = False
+        status_fields['pending'] = False
+        status_fields['reception_control'] = False
+
+        # Tags
+        status_fields['need_review'] = False
+
+        if proj_details.get('aborted'):
+            status_fields['status'] = 'Aborted'
+            status_fields['aborted'] = True
+            status_fields['closed'] = True
+        else:
+            if self.obj.get('close_date'):
+                status_fields['status'] = 'Closed'
+                status_fields['closed'] = True
+            else:
+                if self.obj.get('open_date'):
+                    if self.obj.get('escalations'):
```

Comment:
Maybe all the `status_fields['open'] = True` statements can be substituted for a single one here?

Reply:
Yes it could, but I think it would make it harder to see all the details quickly. Currently, everything that needs to be True is set to True where the 'status' field is set, I think that's quite nice. It would also only save 1 line of code.

Reply:
True that it would only save a single line of code.